app/interactors/rake_interactor.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+require 'rake'
+
+class RakeInteractor
+  include Interactor
+
+  def initialize(task_name: '', args: {}, log_file: nil)
+    @task_name = task_name
+    @args      = args
+    @log_file  = log_file || "#{Rails.root}/log/rake.log"
+
+    super
+  end
+
+  def call
+    old_stdout = $stdout
+    log_buffer = StringIO.new
+    $stdout = log_buffer
```

Comment:
To avoid rewriting all rake tasks to write to a log file, we instead create a temporary buffer to capture all print statements.

This allows us to grab all print statements called within the rake task, which we can then write manually to the logfile of our choice.